tensorflow/core/kernels/uniform_quant_ops/uniform_quantize_op_test.cc

```diff
@@ -59,55 +107,35 @@ TEST_F(UniformQuantizeOpsTest, QuantizeInvalidQuantizationAxis) {
+  std::vector<long> shape{2, 3};
+  QParams qparams(-1, -127, 127, {0.25}, {-20});
+  std::vector<qint8> expected{-127, -100, -20, -16, 0, 20};
+  TestUniformQuantize<float, qint8>(input, shape, qparams, expected);
+}
 
-  AddInputFromArray<float>(TensorShape({2, 3}),
-                           {-27.0, -20.0, 0.0, 1.0, 5.0, 10.0});
-  AddInputFromArray<float>(TensorShape({}), {0.25});
-  AddInputFromArray<int32>(TensorShape({}), {-20});
-
-  TF_ASSERT_OK(RunOpKernel());
-  Tensor expected(allocator(), DT_QINT8, TensorShape({2, 3}));
-  // Input element -27.0 is quantized to -127 (not -128) because
-  // output_quantization_min_val is -127.
-  test::FillValues<qint8>(&expected, {-127, -100, -20, -16, 0, 20});
-  test::ExpectTensorEqual<qint8>(expected, *GetOutput(0));
+TEST_F(UniformQuantizeOpsTest, PerTensorQuint8Quantize) {
+  std::vector<float> input{25.0, 20.5, 0.0, 0.5, 15.0, 13.5};
+  std::vector<long> shape{2, 3};
+  QParams qparams(-1, 0, 255, {0.25}, {-4});
+  std::vector<quint8> expected{96, 78, 0, 0, 56, 50};
+  TestUniformQuantize<float, quint8>(input, shape, qparams, expected);
 }
 
 TEST_F(UniformQuantizeOpsTest, PerChannelQuantize) {
-  TF_ASSERT_OK(NodeDefBuilder("test", "UniformQuantize")
-                   .Input(FakeInput(DT_FLOAT))
-                   .Input(FakeInput(DT_FLOAT))
-                   .Input(FakeInput(DT_INT32))
-                   .Attr("Tin", DT_FLOAT)
-                   .Attr("Tout", DT_QINT8)
-                   .Attr("quantization_axis", 0)
-                   .Attr("quantization_min_val", -127)
-                   .Attr("quantization_max_val", 127)
-                   .Finalize(node_def()));
-  TF_ASSERT_OK(InitOp());
+  std::vector<float> input{-27.0, -20.0, 0.0, 1.0, 5.0, 10.0};
+  std::vector<long> shape{2, 3};
+  QParams qparams(0, -127, 127, {0.25, 0.5}, {-20, -10});
+  std::vector<qint8> expected{-127, -100, -20, -8, 0, 10};
+  TestUniformQuantize<float, qint8>(input, shape, qparams, expected);
+}
 
-  AddInputFromArray<float>(TensorShape({2, 3}),
-                           {-27.0, -20.0, 0.0, 1.0, 5.0, 10.0});
-  AddInputFromArray<float>(TensorShape({2}), {0.25, 0.5});
-  AddInputFromArray<int32>(TensorShape({2}), {-20, -10});
-
-  TF_ASSERT_OK(RunOpKernel());
-  Tensor expected(allocator(), DT_QINT8, TensorShape({2, 3}));
-  // Input element -27.0 is quantized to -127 (not -128) because
-  // output_quantization_min_val is -127.
-  test::FillValues<qint8>(&expected, {-127, -100, -20, -8, 0, 10});
-  test::ExpectTensorEqual<qint8>(expected, *GetOutput(0));
+TEST_F(UniformQuantizeOpsTest, PerChannelQuint8Quantize) {
+  std::vector<float> input{25.0, 20.5, 0.0, 0.5, 15.0, 13.5};
+  std::vector<long> shape{2, 3};
+  QParams qparams(0, 0, 255, {0.25, 0.5}, {-20, -10});
+  std::vector<quint8> expected{80, 62, 0, 0, 20, 17};
+  TestUniformQuantize<float, quint8>(input, shape, qparams, expected);
 }
 
-}  // namespace tensorflow
+}  // namespace tensorflow
```

Comment:
It looks like your editor may be deleting newlines at EoF, can these be reverted? I believe we prefer newline at eof, as it is safer with patch/diff tools

Reply:
Thank you, the new lines at EoF have been restored.